### DISCUSSION TITLE
Is it possibly to specify a java agent

### DISCUSSION THREAD

**Discussion by danielr1996** [itzg/docker-minecraft-server#3023] [Q&A] [answered]
I'd like to use [Grafana Pyroscope](https://grafana.com/docs/pyroscope/latest/configure-client/language-sdks/java/ ) to profile my server, which requires to set the `-javagent` flag. However it seems there is no way of settings this, unline the various `JVM_*` options. Is there a way to completely override the startup command so I can inject the flag, or could we add a JAVA_AGENT environment variable to set this?

**Comment by itzg** [comment, marked as answer]
`JVM_OPTS` is where you would set `-javaagent` argument 

https://docker-minecraft-server.readthedocs.io/en/latest/configuration/jvm-options/#extra-jvm-options

**Reply by danielr1996** [reply]
Oh well, I didn't realize that `JVM_OPTS` allows arbitry options besides `X`, `XX` and `DD`, works perfectly
